descriptive-link-5effbb.md

```diff
@@ -0,0 +1,237 @@
+
+```html
+<p><a href="#" aria-label="Place item in cart">Place item</a><span aria-hidden="true"> in shopping cart</span></p>
+```
+
+#### Passed Example 5
+
+Accessible name describes the purpose of the link
+
+```html
+<div role="link" aria-label="Skip to main content" onclick="document.location+='#main';return false;"></div>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+#### Passed Example 6
+
+Link text describes the purpose of the link
+
+```html
+<span role="link" onclick="document.location+='#desc';return false;">See description of the product</span>
+
+<p id="desc">This product consists of several web pages</p>
+```
+
+#### Passed Example 7
+
+The context afforded by the list and the link text describe the purpose of the links.
+
+```html
+<ul>
+  <li><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">Ulysses</a></li>
+  <li><a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73">EPUB format</a></li>
+  <li><a href="https://www.gutenberg.org/files/4300/4300-0.txt">Plain text</a></li>
+</ul>
+```
+
+#### Passed Example 8
+
+The context afforded by the table header and the link text describe the purpose of the links.
+
+```html
+<table>
+  <tr>
+    <th colspan="3">Ulysses</th>
+  </tr>
+  <tr>
+    <td><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">HTML</a></td>
+    <td><a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73">EPUB</a></td>
+    <td><a href="https://www.gutenberg.org/files/4300/4300-0.txt">Plain text</a></td>
+  </tr>
+</table>
+```
+
+
+### Failed
+
+#### Failed Example 1
+
+Link text, together with the absence of programatically determined link context, does not describe the purpose of the link
+
+```html
+<a href="#desc">More</a>
+
+<p id="desc">This product consists of several web pages</p>
+```
+
+#### Failed Example 2
+
+Accessible name, together with the absence of programatically determined link context, does not describe the purpose of the link
+
+```html
+<a href="#main" aria-label="Go"></a>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+#### Failed Example 3
+
+Link text, together with the absence of programatically determined link context, does not describe the purpose of the link
+
+```html
+<div role="link" onclick="document.location+='#main';return false;">More</div>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+#### Failed Example 4
+
+Accessible name, together with the absence of programatically determined link context, does not describe the purpose of the link
+
+```html
+<div role="link" aria-labelledby="id1" onclick="document.location+='#main';return false;"></div>
+<div id="id1">Go</div>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+
+### Inapplicable 
+
+#### Inapplicable Example 1
+
+`<a>` with its role changed from `link` to another role
```

Comment:
I think we usually use `a` rather than `<a>`.

Also, for the sake of making nice description, I'd prefer "An `a` element with …"